### PR TITLE
feat: Add group detail view

### LINKED ISSUE #315: Add group detail view
As a group member, I want to click on a group name on the „my groups“ page to see an overview page of the group containing the items I can see from this group.

FYI: this story only focuses on the second part of the group detail page - the item overview. There are other stories for the member list and group description and picture as well. 

Detail view

### Acceptance Criteria

- [ ]  Clicking on a group card item on "my groups" redirects me to the detailed view of this group
- [ ]  Items are shown as described in the wireframe (and/or like everywhere else in the application)
- [ ] Only items the group can see are visible

### PR DESCRIPTION
**Pull request by DieKautz**
This will close #315

### CODE REVIEW

**Review by Encotric — APPROVED** [review]

File: app/views/groups/index.html.erb [hunk 1]
```diff
@@ -10,22 +10,14 @@
   <div class="h3">
     <%= render GroupNameComponent.new(group: group) %>
     <% if @user.admin_in?(group)%>
-      <%= link_to " ", edit_group_path(group), class: "badge bg-primary bi bi-pencil-fill" %>
+      <%= link_to " ", edit_group_path(group), class: "badge bg-primary bi bi-pencil-fill", style: "z-index:2; position:relative" %>
```
Comment (kbroschke) [inline app/views/groups/index.html.erb:13]: Z-indexes 0 to 3 are used by bootstrap classes for different states of elements. (See [their approach to z-index scales](https://getbootstrap.com/docs/5.2/extend/approach/#z-index-scales))

Maybe we should use some higher index, like 4 or 10 (the latter one being more satisfying imo).

File: app/views/groups/index.html.erb [hunk 2]
```diff
@@ -10,22 +10,14 @@
...
     <% else %>
-      <%= link_to t(:leave_group), group_leave_path(group), data: {turbo_method: :post, turbo_confirm: "Are you sure?"}, class: "btn btn-primary" %>
+      <%= link_to t(:leave_group), group_leave_path(group), data: {turbo_method: :post, turbo_confirm: "Are you sure?"}, class: "btn btn-primary", style: "z-index:2; position:relative" %>
```
Comment (kbroschke) [inline app/views/groups/index.html.erb:15]: See comment above for z-index.

File: app/views/groups/index.html.erb (hunk 1 above)
Comment (kbroschke) [inline app/views/groups/index.html.erb:13]:
```suggestion
      <%= link_to " ", edit_group_path(group), class: "badge bg-primary bi bi-pencil-fill position-relative", style: "z-index:2" %>
```

File: app/views/groups/index.html.erb (hunk 2 above)
Comment (kbroschke) [inline app/views/groups/index.html.erb:15]:
```suggestion
      <%= link_to t(:leave_group), group_leave_path(group), data: {turbo_method: :post, turbo_confirm: "Are you sure?"}, class: "btn btn-primary position-relative", style: "z-index:2" %>
```

**Review by kbroschke — CHANGES_REQUESTED** [review]

File: app/views/groups/index.html.erb (hunk 1 above)
Comment (DieKautz) [reply app/views/groups/index.html.erb:13]: Is there anything failing right now?
Would 10 render on top of e.g. modals?

File: app/views/groups/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<h1 class="text-center text-break">
+  <%= render GroupNameComponent.new(group: @group) %>
+</h1>
+
+<div class="text-end">
+  <%= mail_to @admin.email, t(:contact_group), class: "btn btn-sm btn-primary" %>
```
Comment (kbroschke) [inline app/views/groups/show.html.erb:6]: This fails if the group has no admin (e.g. group `everyone`).

@Encotric What should be the expected behavior in such cases?

**Review by kbroschke — CHANGES_REQUESTED** [review]

File: app/views/groups/show.html.erb (hunk 3 above)
Comment (kbroschke) [reply app/views/groups/show.html.erb:6]: As stated in #276, the group `everyone` is not shown to the user to be clicked on.

Should we catch this error anyway? E.g., in controller when `@admin` is `null`?

File: app/views/groups/index.html.erb (hunk 1 above)
Comment (kbroschke) [reply app/views/groups/index.html.erb:13]: Right now, nothing is failing. But – that's why I linked the bootstrap documentation – if we ever decide to change the style of something under the button (e.g., on focus) this styling will be shown above the button (z-index 3).

You can also read in the documentation that all other bootstrap classes (including modals) have z-indexes of 1000 and higher.

Maybe we could create a custom CSS class for all buttons that should be on top of stretch links.

File: app/views/groups/show.html.erb (hunk 3 above)
Comment (DieKautz) [reply app/views/groups/show.html.erb:6]: should resolve with [9035a27](https://github.com/hpi-swt2/bookkeeper-portal-red/pull/360/commits/9035a272508245be967d98d9770512e3ffa7e569)

File: app/views/groups/index.html.erb (hunk 1 above)
Comment (DieKautz) [reply app/views/groups/index.html.erb]: [f2447dd](https://github.com/hpi-swt2/bookkeeper-portal-red/pull/360/commits/f2447dd6e257782cc3561ed67c283233041645ed) should resolve this and all other change requests

File: app/views/groups/index.html.erb (hunk 2 above)
Comment (kbroschke) [reply app/views/groups/index.html.erb]: Resolved by f2447dd

File: app/views/groups/index.html.erb (hunk 1 above)
Comment (kbroschke) [reply app/views/groups/index.html.erb]: Resolved by f2447dd

File: app/views/groups/index.html.erb (hunk 2 above)
Comment (kbroschke) [reply app/views/groups/index.html.erb]: Resolved by f2447dd

**Pull request merged**